### DISCUSSION TITLE
Set env variable with custom string

### DISCUSSION THREAD

**Discussion by jighen555** [thunderclient/thunder-client-support#972] [Q&A]
Hello everyone,

it is possible to set an environment variable with a random string with only upper case character, without space and always 12 character?

Thank you!

**Comment by rangav** [comment]
Hi @jighen555 you can write your own `Custom Filter` and apply in Tests. Select `Set Env Variable` option in Tests for setting variable.

https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#filters-in-tests

Custom Filters
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#custom-filters

Let me know if anything is not clear.

**Reply by jighen555** [reply]
I add this script:
  function setEnvTestVariable{
      return "TEST";
  }
  module.exports = [setEnvTestVariable];

And set the variable like this:
![image](https://user-images.githubusercontent.com/104869891/212053135-494a2255-8ae6-46ea-b754-2253366c97aa.png)

But i receive an error:
![image](https://user-images.githubusercontent.com/104869891/212053320-a1db9da8-83b4-45fa-aa2e-ebd514dc35cf.png)

**Reply by rangav** [reply]
you wrote the function syntax wrong, `() is missing`

the correct format below
```js
function setEnvTestVariable() {
      return "TEST";
 }

  module.exports = [setEnvTestVariable];
```

**Reply by jighen555** [reply]
Ok good sorry for my misunderstanding!
